base/cmd.jl

```diff
@@ -481,6 +481,12 @@ function cmd_gen(parsed)
     end
 end
 
+@assume_effects :effect_free :terminates_globally :noub function cmd_gen(
+    parsed::Tuple{Vararg{Tuple{Vararg{Union{String, SubString{String}}}}}}
+)
+    return @invoke cmd_gen(parsed::Any)
+end
```

Comment:
Isn't this `:nothrow` too? If it is, we should probably annotate it here. Doing so would allow the `Cmd` construction to be removed if it's not used. Without this, the effects annotations might only be beneficial in a REPL completion context.

Reply:
Unfortunately not, `cstr` throws an error for strings containing the nul character. We are kind of inconsistent with where we check this though - arguably the check should be done when running a CMD, not upon construction